Auth.Infrastructure/TransactionFactory.cs

```diff
@@ -0,0 +1,12 @@
+﻿using Auth.Core.Common;
+using Microsoft.EntityFrameworkCore.Storage;
+
+namespace Auth.Infrastructure;
+
+public class TransactionFactory(AuthDbContext _context) : ITransactionFactory
```

Comment:
The only concern I have that this factory wont work with transient contexts, if someone changes the default lifetime for it

Reply:
You are correct, thank you. The intention was to create a POC of outbox pattern, so I didn't focus on any other stuff